src/xr/xr-depth-sensing.js

```diff
@@ -0,0 +1,281 @@
+        this._available = false;
+        this.fire('unavailable');
+    }
+};
+
+/**
+ * @function
+ * @name pc.XrDepthSensing#getDepth
+ * @param {number} x - x coordinate of pixel in depth texture.
+ * @param {number} y - y coordinate of pixel in depth texture.
+ * @description Get depth value from depth information in meters. X and Y coordinates are in depth texture space, use {@link pc.XrDepthSensing#width} and {@link pc.XrDepthSensing#height}. This is not using GPU texture, and is a CPU path.
+ * @example
+ * var depth = app.xr.depthSensing.getDepth(x, y);
+ * if (depth !== null) {
+ *     // depth in meters
+ * }
+ * @returns {number|null} Depth in meters or null if depth information is not available.
+ */
+XrDepthSensing.prototype.getDepth = function (x, y) {
+    if (! this._depthInfo)
+        return null;
+
+    return this._depthInfo.getDepth(x, y);
+};
+
+Object.defineProperty(XrDepthSensing.prototype, 'supported', {
+    get: function () {
+        return !! window.XRDepthInformation;
+    }
+});
+
+/**
+ * @name pc.XrDepthSensing#available
+ * @type {boolean}
+ * @description True if depth sensing information is available.
+ * @example
+ * if (app.xr.depthSensing.available) {
+ *     var depth = app.xr.depthSensing.getDepth(x, y);
+ * }
+ */
+Object.defineProperty(XrDepthSensing.prototype, 'available', {
+    get: function () {
+        return this._available;
+    }
+});
+
+Object.defineProperty(XrDepthSensing.prototype, 'width', {
+    get: function () {
+        return this._depthInfo && this._depthInfo.width || 0;
+    }
+});
+
+Object.defineProperty(XrDepthSensing.prototype, 'height', {
+    get: function () {
+        return this._depthInfo && this._depthInfo.height || 0;
+    }
+});
+
+/**
+ * @name pc.XrDepthSensing#texture
+ * @type {pc.Texture}
+ * @description Texture that contains packed depth information. Format of this texture is {@link pc.PIXELFORMAT_L8_A8}. And is UV transformed based on onderlying AR system, which can be normalized using {@link pc.XrDepthSensing#uvMatrix}.
+ * @example
+ * material.diffuseMap = depthSensing.texture;
+ * @example
+ * // // GLSL shader to unpack depth texture
+ * // varying vec2 vUv0;
+ * //
+ * // uniform sampler2D texture_depthSensingMap;
+ * // uniform mat4 matrix_depth_uv;
+ * //
+ * // void main(void) {
+ * //     // transform UVs using depth matrix
+ * //     vec2 texCoord = (matrix_depth_uv * vec4(vUv0.xy, 0.0, 1.0)).xy;
+ * //
+ * //     // get luminance alpha components from depth texture
+ * //     vec2 packedDepth = texture2D(texture_depthSensingMap, texCoord).ra;
+ * //
+ * //     // unpack into single value in millimeters
+ * //     float depth = dot(packedDepth, vec2(255.0, 256.0 * 255.0)); // mm
+ * //
+ * //     // normalize: 0m to 8m distance
+ * //     depth = min(depth / 8000.0, 1.0); // 0..1 = 0m..8m
+ * //
+ * //     // paint scene from black to white based on distance
+ * //     gl_FragColor = vec4(depth, depth, depth, 1.0);
+ * // }
```

Comment:
Again, how come the example code is commented out? Is this because the linter complains? If so, you can disable the linter for this block with hints. Just search the codebase for `eslint-disable`.

Reply:
Nice one, checked example in existing code, and used it.